### PR TITLE
Respect Ratelimit variables in context

### PR DESCRIPTION
**Pull request by shadialtarsha**
**What does this PR do?**
Right now Falco doesn't know about the penaltybox and ratecounter variables that are declared. 
The linter will throw the error "undefined variables" for them.
This PR aims to fix that.

Also this PR adds support for predefined variables that Fastly provides for the ratelimiter :)

### CODE REVIEW

File: __generator__/builtin.yml [hunk 1]
```diff
@@ -1221,14 +1221,14 @@ ratelimit.check_rate:
   reference: "https://developer.fastly.com/reference/vcl/functions/rate-limiting/ratelimit-check-rate/"
   on: [RECV, HASH, HIT, MISS, PASS, FETCH, ERROR, DELIVER, LOG]
   arguments:
-    - [STRING, ID, INTEGER, INTEGER, INTEGER, ID, TIME]
+    - [STRING, ID, INTEGER, INTEGER, INTEGER, ID, RTIME]
```
Comment (ysugimoto) [inline __generator__/builtin.yml:1224]: Fastly document says the sixth argument type of ttl must be `TIME` type.
https://developer.fastly.com/reference/vcl/functions/rate-limiting/ratelimit-check-rate/
What does this change intend to do?

File: __generator__/builtin.yml [hunk 2]
```diff
@@ -1221,14 +1221,14 @@ ratelimit.check_rate:
...
   return: BOOL
 
 ratelimit.check_rates:
   reference: "https://developer.fastly.com/reference/vcl/functions/rate-limiting/ratelimit-check-rates/"
   on: [RECV, HASH, HIT, MISS, PASS, FETCH, ERROR, DELIVER, LOG]
   arguments:
-    - [STRING, ID, INTEGER, INTEGER, INTEGER, ID, INTEGER, INTEGER, INTEGER, ID, TIME]
+    - [STRING, ID, INTEGER, INTEGER, INTEGER, ID, INTEGER, INTEGER, INTEGER, ID, RTIME]
```
Comment (ysugimoto) [inline __generator__/builtin.yml:1231]: Fastly document says the 11th argument type of ttl must be `TIME` type.
https://developer.fastly.com/reference/vcl/functions/rate-limiting/ratelimit-check-rates/
What does this change intend to do?

File: context/context.go [hunk 3]
```diff
@@ -185,6 +185,31 @@ func (c *Context) GetRegexGroupVariable(name string) (types.Type, error) {
...
+	// nameComponents[0] should be "ratecounter"
+	// nameComponents[1] should be the variable name
+	// nameComponents[2] should be either "bucket" or "rate"
+	// nameComponents[3] should be the time (10s, 60s, etc)
+	if v, ok := c.Variables[nameComponents[0]].Items["%any%"].Items[nameComponents[2]].Items[nameComponents[3]]; ok {
+		return v.Value.Get, nil
+	}
+
+	return types.NullType, fmt.Errorf(`undefined variable "%s"`, name)
+}
```
Comment (ysugimoto) [inline context/context.go:211]: Nice implementation 👍 
If this implementation is available, can we remove 
https://github.com/ysugimoto/falco/blob/main/context/context.go#L206-L210 ?

**Review by ysugimoto — APPROVED** [review]
LGTM with nits 👍

File: __generator__/builtin.yml (hunk 2 above)
Comment (shadialtarsha) [reply __generator__/builtin.yml:1231]: I reached out to Fastly and they confirmed it is a documentation bug. The value should be RTIME

File: __generator__/builtin.yml (hunk 1 above)
Comment (shadialtarsha) [reply __generator__/builtin.yml:1224]: I reached out to Fastly and they confirmed it is a documentation bug. The value should be RTIME

File: context/context.go (hunk 3 above)
Comment (shadialtarsha) [reply context/context.go:211]: yep 
on it

**Review by ysugimoto — APPROVED** [review]
Looks good, nice work!

**Pull request merged**